File: app/src/main/java/com/aa/androidormcomparison/measures/MeasureDatabase/DatesConverter.java
package com.aa.androidormcomparison.measures.MeasureDatabase;

import androidx.room.TypeConverter;

import java.util.Date;

public class DatesConverter {
     @TypeConverter
    public static Date toDate(Long timestamp) {
        return timestamp == null ? null : new Date(timestamp);
    }

    @TypeConverter
    public static Long toTimestamp(Date date) {
        return date == null ? null : date.getTime();
    }
}

File: app/src/main/java/com/aa/androidormcomparison/measures/MeasureDatabase/DatesConverter.kt
package com.aa.androidormcomparison.measures.MeasureDatabase

import androidx.room.TypeConverter
import java.util.*

class DatesConverter {

    @TypeConverter
    fun fromDateToTimestamp(date: Date?):Long?{
        return date?.time
    }

    @TypeConverter
    fun fromTimestampToDate(timestamp: Long?):Date?{
        return if (timestamp == null) null else Date(timestamp)
    }

}